navigator/backend/src/main/scala/com/digitalasset/navigator/model/PackageRegistry.scala

```diff
@@ -51,19 +51,23 @@ case class PackageRegistry(
         p.packageId -> DamlLfPackage(p.packageId, typeDefs, templates)
       }
 
-    val newTemplates = newPackages
-      .map(_._2.templates)
-      .reduce(_ ++ _)
-
-    val newTypeDefs = newPackages
-      .map(_._2.typeDefs)
-      .reduce(_ ++ _)
-
-    copy(
-      packages = packages ++ newPackages,
-      templates = templates ++ newTemplates,
-      typeDefs = typeDefs ++ newTypeDefs
-    )
+    if (newPackages.nonEmpty) {
```

Comment:
Low importance: perhaps an early return would be easier to read?

```scala
if (newPackages.isEmpty) {
  return this
}
```

Reply:
If thought that the use of `return` is discouraged in Scala. It would be easier to read though.